v2/internal/controllers/crd_networking_privatednszone_test.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"testing"
+
+	network "github.com/Azure/azure-service-operator/v2/api/network/v1beta20180901"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/go-autorest/autorest/to"
+	. "github.com/onsi/gomega"
+)
+
+func Test_Networking_PrivateDnsZone_CRUD(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	name := tc.Namer.GenerateName("pdz") + ".com"
+	zone := &network.PrivateDnsZone{
+		ObjectMeta: tc.MakeObjectMetaWithName(name),
+		Spec: network.PrivateDnsZones_Spec{
+			Location: to.StringPtr("global"),
+			Owner:    testcommon.AsOwner(rg),
+		},
+	}
+
+	tc.CreateResourceAndWait(zone)
+
+	tc.Expect(zone.Status.MaxNumberOfRecordSets).ToNot(BeNil())
+	tc.Expect(zone.Status.MaxNumberOfVirtualNetworkLinks).ToNot(BeNil())
+
+	tc.DeleteResourceAndWait(zone)
```

Comment:
Minor: Could add assertion after this that delete really happened in Azure. Look for example tests that use the `HEAD` API to do that. For example the `crd_cache_redis_test.go` does that

Reply:
Checking against Azure requires an ID or something to query for, and there's nothing on the object. Other tests use an Id property that doesn't exist here.

Reply:
We also compute the Azure ID and store it in an annotation: `serviceoperator.azure.com/resource-id`, which is accessible as `genruntime.ResourceIDAnnotation` or readable via `genruntime.GetResourceID(obj)`

Reply:
Added.